### PR TITLE
Add cli changes for windows GPU feature

### PR DESCRIPTION
**Pull request by pineapplethevoyager**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
az aks nodepool add

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/aks-preview/HISTORY.rst
Comment (FumingZhang) [inline src/aks-preview/HISTORY.rst:31]: Move your change history to section Pending (wait until someone else releases a new version including your changes later)  OR

pick a new version number (and update version in src/aks-preview/setup.py) to release the change once the PR got merged

**Review by FumingZhang — COMMENTED** [review]
Please add a live test case for the new option

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py
Comment (chengliangli0918) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:3626]: line with snapshot_name can be removed.

File: src/aks-preview/HISTORY.rst
Comment (chengliangli0918) [inline src/aks-preview/HISTORY.rst:15]: Check with @FumingZhang  if you need to add a new version or add under 'Pending' section.

Comment (pineapplethevoyager) [reply src/aks-preview/HISTORY.rst:15]: I had added a new version number he gave us the option to do either

File: src/aks-preview/setup.py
Comment (pineapplethevoyager) [reply src/aks-preview/setup.py:12]: Yes see his above comment for details

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:3613]: Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=85473147&view=results) to validate the change.

**Review by FumingZhang — COMMENTED** [review]
Please fix failed style check

could fix this by adding a comment `# pylint: disable=no-member`
> src/aks-preview/azext_aks_preview/agentpool_decorator.py:575:40: E1101: Instance of 'AKSPreviewAgentPoolModels' has no 'AgentPoolGPUProfile' member (no-member)

> ERROR: /mnt/vss/_work/1/s/src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:3661:17: E128 continuation line under-indented for visual indent

**Comment by FumingZhang** [comment]
Wow, there are too many changes here that are not related to the feature. Please rebase from main branch rather than merge the main branch.

And there's a comment thread discussing about the new version naming pattern. See https://github.com/Azure/azure-cli-extensions/pull/7151#discussion_r1446925040

File: src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py
Comment (FumingZhang) [reply src/aks-preview/azext_aks_preview/tests/latest/test_aks_commands.py:3613]: Test passed!

File: src/aks-preview/HISTORY.rst
Comment (FumingZhang) [inline src/aks-preview/HISTORY.rst:15]: Please resolve conflict and choose a new version number like `1.0.0b2`

File: src/aks-preview/HISTORY.rst
Comment (FumingZhang) [inline src/aks-preview/HISTORY.rst:18]: This change is actually released in 1.0.0b1

Comment (chengliangli0918) [reply src/aks-preview/HISTORY.rst:18]: @FumingZhang do you want use to move the '--skip-gpu-driver-install' change to the section under 1.0.0b1?

File: src/aks-preview/setup.py
Comment (chengliangli0918) [inline src/aks-preview/setup.py:12]: @FumingZhang also no version bump here?

File: src/aks-preview/HISTORY.rst
Comment (FumingZhang) [reply src/aks-preview/HISTORY.rst:18]: no

I mean the change "Add `--ip-families` to the `az aks update` command." is acutally released in `1.0.0b1`. Please help correct the history note by moving this to 1.0.0b1. You could keep the history note of your change in 1.0.0b2 and release a new version.

File: src/aks-preview/setup.py
Comment (FumingZhang) [reply src/aks-preview/setup.py:12]: I suppose it's ok to bump the version here to trigger a new release

File: src/aks-preview/HISTORY.rst
Comment (chengliangli0918) [reply src/aks-preview/HISTORY.rst]: @FumingZhang got you, I made the corresponding change according to your comment because @pineapplethevoyager  is OOF today.

**Review by FumingZhang — APPROVED** [review]
LGTM

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**
